### PR TITLE
Device interaction

### PR DESCRIPTION
**Pull request by meriadec**
interaction

This PR include lot of side-stuff impacted by the DeviceInteraction component.

- Rewrite of the GenuineCheck/GenuineCheckModal
- Rewrite of the EnsureDeviceApp component
- Rewrite of the ReceiveModal

As design has been made by *your servitor*, it may causes some iterations with design team. Anyway, let's ship, and polish later.

:rocket:

### CODE REVIEW

File: src/components/DeviceInteraction/DeviceInteractionStep.js [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+      this.run()
+    }
+  }
+
+  state = {
+    status: 'idle',
+  }
+
+  componentDidMount() {
+    this.__IS_MOUNTED__THX_FOR_REMOVING_COMPONENTWILLMOUNT__ = true
```
Comment (gre) [inline src/components/DeviceInteraction/DeviceInteractionStep.js]: which mean you don't actually need to do this here 🤔

File: src/components/DeviceInteraction/DeviceInteractionStep.js [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+    const { onFail, step } = this.props
+    this.setState({ status: 'idle' })
+    onFail(e, step)
+  }
+
+  run = async () => {
+    const { step, data } = this.props
+
+    if (this.__IS_MOUNTED__THX_FOR_REMOVING_COMPONENTWILLMOUNT__) {
+      this.setState({ status: 'running' })
```
Comment (gre) [inline src/components/DeviceInteraction/DeviceInteractionStep.js]: in the init case, we actually are already in running case 🤔

Comment (meriadec) [reply src/components/DeviceInteraction/DeviceInteractionStep.js]: yeah but we don't want to `setState` if we already are running.
nvm refacto incoming

**Comment by meriadec** [comment]
@gre I think we can merge. I pixel push tomorrow w design.

**Comment by gre** [comment]
let's do this

**Pull request merged**